pal/input/web/gamepad-input.ts

```diff
@@ -225,42 +227,52 @@ export class GamepadInputDevice {
         return device;
     }
 
-    private static _ensureDirectorDefined (): Promise<void> {
-        return new Promise<void>((resolve) => {
-            GamepadInputDevice._intervalId = setInterval(() => {
-                if (legacyCC.director && legacyCC.Director) {
-                    clearInterval(GamepadInputDevice._intervalId);
-                    GamepadInputDevice._intervalId = -1;
-                    resolve();
-                }
-            }, 50);
-        });
+    private static _ensureDirectorDefined (callback: () => void): void {
+        GamepadInputDevice._intervalId = setInterval(() => {
+            if (legacyCC.director && legacyCC.Director) {
+                clearInterval(GamepadInputDevice._intervalId);
+                GamepadInputDevice._intervalId = -1;
+                callback();
+            }
+        }, 50);
+    }
+
+    private static _totalGamepadCnt = 0;
+    private static _updateGamepadCnt (): void {
+        GamepadInputDevice._totalGamepadCnt = GamepadInputDevice._cachedWebGamepads.reduce((total, gamepad) => (gamepad ? total + 1 : total), 0);
     }
 
     private static _registerEvent (): void {
-        GamepadInputDevice._ensureDirectorDefined().then(() => {
+        GamepadInputDevice._ensureDirectorDefined(() => {
+            GamepadInputDevice._cachedWebGamepads = GamepadInputDevice._getWebGamePads();
+            GamepadInputDevice._updateGamepadCnt();
             legacyCC.director.on(legacyCC.Director.EVENT_BEGIN_FRAME, GamepadInputDevice._scanGamepads);
-        }).catch((e) => {});
+        });
         window.addEventListener('gamepadconnected', (e) => {
             GamepadInputDevice._cachedWebGamepads[e.gamepad.index] = e.gamepad;
+            GamepadInputDevice._updateGamepadCnt();
```

Comment:
The same as above.

Reply:
I think this approach ensures that the value of _totalGamepadCnt is always accurate, even if there are changes to _cachedWebGamepads elsewhere in the code.